content/blog/2020-06-28-shtab-completion.md

```diff
@@ -0,0 +1,197 @@
+---
+title: '(Tab) Complete Any Python Application in 1 Minute or Less'
+date: 2020-06-28
+description: |
+  We've made a painless tab-completion script generator for Python applications!
+  Find out how to take advantage of it in this blog post.
+
+descriptionLong: |
+  We've made a painless tab-completion script generator for Python applications!
+  It's called `shtab` and it currently works with `argparse`, `docopt`, and
+  `argopt` to produce `bash` and `zsh` completion scripts.
+  This tool was originally created to help `dvc`, but we realised it could be made
+  more generic and valuable to the world's entire ecosystem of Python CLI
+  applications. Find out how to take advantage of it in this blog post.
+author: casper_dcl
+---
+
+> Other potential titles:
+>
+> - Zero Effort Tab Completion for Python Applications
+> - Finally: 1-Click Tab Completion for [`argparse`], [`docopt`] and [`argopt`]
+> - A [`shtab`] in the dark at completing your ~~sentences~~ commands
+
+Command line tools are powerful. Things like [`make`] have manual pages
+spanning, well,
+[pages](https://www.gnu.org/software/make/manual/make.html#Options-Summary),
+while just the list of [`git`] subcommands is longer than can fit on a standard
+`80 x 24` terminal screen.
+
+```dvc
+$ git <TAB>
+add                  filter-branch        rebase
+am                   format-patch         reflog
+annotate             fsck                 relink
+...
+describe             prco                 unassume
+--More--
+```
+
+Notice the `--More--` at the bottom? That's the joy of pagination.
+
+Notice the `<TAB>` at the top? That represents actually pressing the tab key.
+Ah, the joy of shell tab completion.
+
+Tab completion is an indispensable part of writing anything on the command-line.
+Personally, I can't imaging trying to `git co` (aliased to `git checkout`) a
+branch without `<TAB>` to do the heavy lifting.
+
+Now there's a tool called `dvc` which is like [Git for data](https://dvc.org).
+It can be viewed as a cross-platform combination of [`git`] and [`make`]
+designed for handling big data and multiple cloud storage repositories, as well
+as tracking machine learning experiments. As you can imagine, supporting that
```

Comment:
Since we're in dvc.org/blog, not sure the description of DVC is needed 🙂 but OK up to you.

Reply:
not intended to be merged here - this PR is mainly to make it easy to review before publishing elsewhere...